### PR TITLE
[App] Pass LightningWork to LightningApp

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This feature was requested by @williamFalcon and enables passing work to the LightningApp as well as a flow.

This meant mainly removing boilerplate from documentation examples.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton @rohitgr7

### CODE REVIEW

File: src/lightning_app/core/flow.py [hunk 1]
```diff
@@ -763,3 +763,15 @@ def load_state_dict(self, flow_state, children_states, strict) -> None:
...
+class _RootFlow(LightningFlow):
+    def __init__(self, work):
+        super().__init__()
+        self.work = work
+
+    def run(self):
+        self.work.run()
+
+    def configure_layout(self):
+        return [{"name": "Main", "content": self.work}]
```
Comment (hhsecond) [inline src/lightning_app/core/flow.py:777]:
```suggestion
        return [{"name": "Main", "content": self.work.url}]
```

File: src/lightning_app/core/app.py [hunk 2]
```diff
@@ -62,8 +62,8 @@ def __init__(
         the :class:`~lightning_app.core.flow.LightningFlow` provided.
 
         Arguments:
-            root: The root LightningFlow component, that defines all the app's nested components, running infinitely.
-                It must define a `run()` method that the app can call.
+            root: The root LightningFlow or LightningWork component, that defines all the app's nested components,
```
Comment (Borda) [inline src/lightning_app/core/app.py:65]:
```suggestion
            root: The root `LightningFlow` or `LightningWork` component, that defines all the app's nested components,
```

**Review by lantiga — APPROVED** [review]
LGTM!

File: src/lightning_app/core/flow.py (hunk 1 above)
Comment (hhsecond) [reply src/lightning_app/core/flow.py:777]: or does the `content` fetch that automatically?

File: tests/tests_app/core/test_lightning_work.py [hunk 3]
```diff
@@ -327,3 +328,21 @@ def run(self, *args, **kwargs):
...
+
+class LightningTestAppWithWork(LightningTestApp):
+    def on_before_run_once(self):
+        if self.root.work.has_succeeded:
+            return True
+        return super().on_before_run_once()
+
+
+def test_lightning_app_with_work():
+
```
Comment (Borda) [inline tests/tests_app/core/test_lightning_work.py:346]:
```suggestion
```

File: src/lightning_app/core/flow.py (hunk 1 above)
Comment (tchaton) [reply src/lightning_app/core/flow.py:777]: Yes, it gets the URL directly ;)  You can pass the url only if you want to add more than the root path.

File: src/lightning_app/CHANGELOG.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added support for adding descriptions to commands either through a docstring or the `DESCRIPTION` attribute ([#15193](https://github.com/Lightning-AI/lightning/pull/15193)
 - Added a try / catch mechanism around request processing to avoid killing the flow ([#15187](https://github.com/Lightning-AI/lightning/pull/15187)
 - Added a Database Component ([#14995](https://github.com/Lightning-AI/lightning/pull/14995)
-
+- Added support to pass a LightningWork to the LightningApp ([#15215](https://github.com/Lightning-AI/lightning/pull/15215)
```
Comment (Borda) [inline src/lightning_app/CHANGELOG.md:20]:
```suggestion
- Added support to pass a `LightningWork` to the `LightningApp` ([#15215](https://github.com/Lightning-AI/lightning/pull/15215)
```

**Review by Borda — APPROVED** [review]

**Review by hhsecond — APPROVED** [review]
